packages/medusa/src/services/payment-collection.ts

```diff
@@ -365,6 +365,28 @@ export default class PaymentCollectionService extends TransactionBaseService {
     })
   }
 
+  async markAsAuthorized(
+    paymentCollectionId: string
+  ): Promise<PaymentCollection> {
+    return await this.atomicPhase_(async (manager) => {
+      const paymentCollectionRepo = manager.getCustomRepository(
+        this.paymentCollectionRepository_
+      )
+
+      const paymentCollection = await this.retrieve(paymentCollectionId)
+      paymentCollection.status = PaymentCollectionStatus.AUTHORIZED
+      paymentCollection.authorized_amount = paymentCollection.amount
+
+      const result = await paymentCollectionRepo.save(paymentCollection)
+
+      await this.eventBusService_
+        .withTransaction(manager)
+        .emit(PaymentCollectionService.Events.UPDATED, result)
```

Comment:
Hmm, that's a good question. If we don't have a payment, we are throwing 404s on all payments operations post-authorization right?

For draft orders, we create a "system payment" using the default payment provider in the store, it might make sense to follow the same approach.

What's your take?

Reply:
> The subsequent payment operations (capture or refund) of a Payment collection would required the payment.id that wouldn't exist in this case

Yes, this is the scenarios I highlighted. That would throw 404s. Do you find that misleading or should we assume that the consumer is known to the fact that the payment was forced through?